lib/matplotlib/backends/backend_agg.py

```diff
@@ -527,7 +563,14 @@ def print_png(self, filename_or_obj, *args, **kwargs):
         else:
             close = False
         try:
-            _png.write_png(renderer._renderer, filename_or_obj, self.figure.dpi)
+            if kwargs.get('flatten', False):
+                w, h = int(renderer.width), int(renderer.height)
+                img = renderer._renderer.buffer_rgba()
+                img = np.array(memoryview(img)).reshape((h,w,4))
+                img = self.flatten_rgba(img)
+                _png.write_png(img, filename_or_obj, self.figure.dpi)
+            else:
+                _png.write_png(renderer._renderer, filename_or_obj, self.figure.dpi)
```

Comment:
@mdboom I really wanted to do that at first, but I will be honest when I say that I didn't understand what was going on here: https://github.com/matplotlib/matplotlib/blob/0037595096e3be3ec863f3e3008e73c469e2f2b8/src/_backend_agg.cpp#L28.  I wasn't exactly sure if `write_png` needed something from that or not.